src/JustSaying/Fluent/TopicPublicationBuilder`1.cs

```diff
@@ -71,6 +77,43 @@ public TopicPublicationBuilder<T> WithWriteConfiguration(Action<SnsWriteConfigur
             return this;
         }
 
+        /// <summary>
+        /// Creates a tag with no value that will be assigned to the SNS topic.
+        /// </summary>
+        /// <param name="key">The key for the tag.</param>
+        /// <returns>
+        /// The current <see cref="TopicPublicationBuilder{T}"/>.
+        /// </returns>
+        /// <remarks>Tag keys are case-sensitive. A new tag with a key identical to that of an existing one will overwrite it.</remarks>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="key"/> is <see langword="null"/> or whitespace.
+        /// </exception>
+        public TopicPublicationBuilder<T> WithTag(string key) => WithTag(key, null);
+
+        /// <summary>
+        /// Creates a tag with a value that will be assigned to the SNS topic.
+        /// </summary>
+        /// <param name="key">The key for the tag.</param>
+        /// <param name="value">The value associated with this tag.</param>
+        /// <returns>
+        /// The current <see cref="TopicPublicationBuilder{T}"/>.
+        /// </returns>
+        /// <remarks>Tag keys are case-sensitive. A new tag with a key identical to that of an existing one will overwrite it.</remarks>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="key"/> is <see langword="null"/> or whitespace.
+        /// </exception>
+        public TopicPublicationBuilder<T> WithTag(string key, string value)
+        {
+            if (string.IsNullOrWhiteSpace(key))
+            {
+                throw new ArgumentNullException(nameof(key));
```

Comment:
Shouldn't really throw `ArgumentNullException` if `key` isn't exactly `null`. Should either use `ArgumentException` for both cases and an add an appropriate message, or use `ArgumentNullException` for `null` and `ArgumentException` for white space.